### DISCUSSION TITLE
download mods instead of getting disconnected

### DISCUSSION THREAD

**Discussion by shayanshokry** [efroemling/ballistica#427] [Ideas]
is there any way to download the mods, characters, maps... from the server instead of getting disconnected? its boring to download the exact same mods and trying to connect to the modded server :(

**Comment by bseditor** [comment]
There is a server-side way. If the server-owner uploads the required mods to a file shareable site and gave its link as a STATS button.

**Comment by IvanPragma** [comment]
In Bomb Squad 1.8, this problem will be solved. The new dependency system will allow you to dynamically download assets when you join to the server. https://github.com/efroemling/ballistica/wiki/Dependency-System

**Comment by efroemling** [comment]
Yes, the asset-package system coming up in 1.8 is intended to solve this exact problem. It will let the game automatically grab all the assets (textures, models, etc.) that you need to join modded servers.
Note that this explicitly will *not* include Python scripts, so you'll still need to grab those in some other way if you want to *host* using a given mod. This is for security reasons - downloading & running arbitrary Python code from a random server you are joining would be a very bad idea. But perhaps we can come up with some sort of system to share pre-approved safe mod scripts. ([see this discussion](https://github.com/efroemling/ballistica/discussions/451))
